tools/buildgen/plugins/transitive_dependencies.py

```diff
@@ -16,43 +16,49 @@
 This takes the list of libs, node_modules, and targets from our
 yaml dictionary, and adds to each the transitive closure
 of the list of dependencies.
-
 """
 
 
-def get_lib(libs, name):
-    try:
-        return next(lib for lib in libs if lib['name'] == name)
-    except StopIteration:
-        return None
+def transitive_deps(lib_map, node):
+    """Returns a list of transitive dependencies from node.
+
+    Recursively iterate all dependent node in a depth-first fashion and
+    list a result using a topological sorting.
+    """
+    result = []
+    seen = set()
+    start = node
 
+    def recursive_helper(node):
+        if node is None:
+            return
+        for dep in node.get("deps", []):
+            if dep not in seen:
+                seen.add(dep)
+                next_node = lib_map.get(dep)
+                if next_node:
```

Comment:
Based on your base case, I think this conditional is redundant.

Reply:
Right. Good catch!